client/router.js

```diff
@@ -352,14 +381,51 @@ Router.prototype._invalidateTracker = function() {
 Router.prototype._updateCallbacks = function () {
   var self = this;
 
-  // add global middleware
   self._page.callbacks = [];
+  self._page.exits = [];
+
+  // add global enter triggers
+  self._page("*", function(ctx, next) {
+    _.each(self._triggersEnter, function(fn) {
+      if (typeof fn === 'function') {
+        fn(ctx);
+      }
+    });
+
+    next();
+  });
+
+  // add global exit triggers
+  self._page.exit("*", function(ctx, next) {
```

Comment:
`updateCallback` function seems pretty lengthy now. How abotu moving breaking these into multiple functions. We can write it like this.

~~~
self._page("*", self._processTriggersBefore.bind(self))
self._page.exit("*", self._processTriggersExit.bind(self))
~~~